sympy/printing/tests/test_maple.py

```diff
@@ -0,0 +1,387 @@
+    assert maple_code(A[:, 0]) == \
+           'Matrix([[1], [0], [0]], storage = rectangular)'
+    assert maple_code(A[0, :]) == \
+           'Matrix([[1, sin((1/2)*x), abs(x)]], storage = rectangular)'
+    assert maple_code(Matrix([[x, x - y, -y]])) == \
+           'Matrix([[x, x - y, -y]], storage = rectangular)'
+
+    # empty matrices
+    assert maple_code(Matrix(0, 0, [])) == \
+           'Matrix([], storage = rectangular)'
+    assert maple_code(Matrix(0, 3, [])) == \
+           'Matrix([], storage = rectangular)'
+
+
+def test_vector_entries_hadamard():
+    # For a row or column, user might to use the other dimension
+    A = Matrix([[1, sin(2/x), 3*pi/x/5]])
+    assert maple_code(A) == \
+           'Matrix([[1, sin(2/x), (3/5)*Pi/x]], storage = rectangular)'
+    assert maple_code(A.T) == \
+           'Matrix([[1], [sin(2/x)], [(3/5)*Pi/x]], storage = rectangular)'
+
+
+def test_Matrices_entries_not_hadamard():
+    # For Matrix with col >= 2, row >= 2, they need to be scalars
+    # FIXME: is it worth worrying about this?  Its not wrong, just
+    # leave it user's responsibility to put scalar data for x.
+    A = Matrix([[1, sin(2/x), 3*pi/x/5], [1, 2, x*y]])
+    expected = \
+        'Matrix([[1, sin(2/x), (3/5)*Pi/x], [1, 2, x*y]], ' \
+        'storage = rectangular)'
+    assert maple_code(A) == expected
+
+
+def test_MatrixSymbol():
+    n = Symbol('n', integer=True)
+    A = MatrixSymbol('A', n, n)
+    B = MatrixSymbol('B', n, n)
+    assert maple_code(A*B) == "A.B"
+    assert maple_code(B*A) == "B.A"
+    assert maple_code(2*A*B) == "2*A.B"
+    assert maple_code(B*2*A) == "2*B.A"
+
+    assert maple_code(A*(B + 3*Identity(n))) == "A*(3*Matrix([[1, 0, 0], [0, 1, 0], [0, 0, 1]], storage = sparse) + B)"
+    assert maple_code(A ** (x ** 2)) == "A^(x^2)"
+    assert maple_code(A ** 3) == "A^3"
+    assert maple_code(A ** (S.Half)) == "A^(1/2)"
+
+
+def test_special_matrices():
+    assert maple_code(6*Identity(3)) == "6*Matrix([[1, 0, 0], [0, 1, 0], [0, 0, 1]], storage = sparse)"
+
+
+def test_containers():
+    assert maple_code([1, 2, 3, [4, 5, [6, 7]], 8, [9, 10], 11]) == \
+           "[1, 2, 3, [4, 5, [6, 7]], 8, [9, 10], 11]"
+
+    # FIXME: need more info about tuple in maple
+    assert maple_code((1, 2, (3, 4))) == "[1, 2, [3, 4]]"
+    assert maple_code([1]) == "[1]"
+    assert maple_code((1,)) == "[1]"
+    assert maple_code(Tuple(*[1, 2, 3])) == "[1, 2, 3]"
+    assert maple_code((1, x*y, (3, x ** 2))) == "[1, x*y, [3, x^2]]"
+    # scalar, matrix, empty matrix and empty list
+    assert maple_code((1, eye(3), Matrix(0, 0, []), [])) == \
+           "[1, Matrix([[1, 0, 0], [0, 1, 0], [0, 0, 1]], storage = sparse), zeros(0, 0), Matrix([], storage = rectangular)]"
+
+    print("Need more info about tuple in maple.")
+
+
+# There possibly no such feature in maple
+r"""
+def test_maple_noninline():
+    # FIXME: need more info.
+    source = maple_code((x + y)/Catalan, assign_to='me', inline=False)
```

Comment:
Probably `Catalan` in this test should be replaced with `GoldenRatio` and then it should print (something) like:

```
GoldenRatio := (1/2 + (1/2)*sqrt(5))
me := (x + y)/GoldenRatio
```

I think the purpose is somehow to define undefined symbols on separate lines rather than printing:
```
me := (x + y)/(1/2 + (1/2)*sqrt(5))
```

Not sure how the printer should know that it is a non-trivial expression though.

Reply:
Looking at the Julia printer, it seems like they have separate support for it:

https://github.com/sympy/sympy/blob/de0a1a72e45bf1ead69b527a2c798e59372758e2/sympy/printing/julia.py#L250-L254

However, I do not think that is required in Maple. Of course, if you use lots of GoldenRatio in your expression, it will be more compact in this way. On the other hand, Maple will still replace it once defining me (if I remember correctly), so it will not make any difference later.

Just skip it, I would say.